verilog/CST/declaration.h

```diff
@@ -69,11 +69,20 @@ verible::SymbolPtr MakeInstantiationBase(T1&& type, T2&& decl_list) {
                                  std::forward<T2>(decl_list));
 }
 
+// Remaps an instantiation and extension based call to resemble
+// proper funciton call strtucture
+verible::SymbolPtr ReshapeExtendedFunctionCallInDataDeclaration(
+    verible::SymbolPtr* function_call, verible::SymbolPtr* semicolon);
+
 // Interface for consistently building a data declaration.
 template <typename T1, typename T2, typename T3>
 verible::SymbolPtr MakeDataDeclaration(T1&& qualifiers, T2&& inst_base,
                                        T3&& semicolon) {
   verible::CheckOptionalSymbolAsNode(qualifiers, NodeEnum::kQualifierList);
+  if (inst_base.get()->Tag().tag == (int)NodeEnum::kFunctionCall) {
+    return ABSL_DIE_IF_NULL(
```

Comment:
I think there is no way that this would trigger, at least not before changing the second `DIE_IF_NULL`. But if it should never happen, there is little to no harm in changing it to something else. Both clauses were substituted such that they still should never encounter a malformed clause but in a case that it happens it should not immediately die.